readiness-tool/popup.js

```diff
@@ -0,0 +1,238 @@
+  supportedAds.appendChild(
+    makeList(detectedApps.supported.ads, false, listAllApps));
+  supportedAnalytics.appendChild(
+    makeList(detectedApps.supported.analytics, false, listAllApps));
+  notSupportedAds.appendChild(
+    makeList(detectedApps.notSupported.ads, true, listAllApps));
+  notSupportedAnalytics.appendChild(
+    makeList(detectedApps.notSupported.analytics, true, listAllApps));
+}
+/**
+ * Splits all detected apps into 'supported' and 'not supported'
+ * @param {String} htmlString - String containing all HTML on the page
+ * @param {!Object} listAllApps - JSON of all the 3p vendors 
+ * @return {Object} 
+ */
+function filteredApps(htmlString, listAllApps) {
+  const foundThis = {
+    'supported': {
+      'ads': [],
+      'analytics': []
+    },
+    'notSupported': {
+      'ads': [],
+      'analytics': []
+    }
+  };
+  // for all the app objects in the apps.JSON file
+  Object.keys(listAllApps).forEach(function (key) {
+    let val = listAllApps[key];
+    // If object has a 'script' key
+    if (val.script != null) {
+      Object.keys(val.script).forEach(function (x) {
+        addToDict(val.script[x]
+                  .split('\\;')[0], htmlString, foundThis, key, val.cats[0]);
+      });
+    }
+  });
+  return foundThis;
+}
+/**
+ * Pushes keys to the supported or not supported list of the object
+ 'found this'
+ * @param {String} regexString - String representation of regular expression
+ * @param {!String} htmlString - String containing all HTML on the page
+ * @param {Object} foundThis - Object separating the 3P services by support
+ * @param {String} key - name of third party service
+ * @param {String} category - the category that the key belongs to
+ */
+function addToDict(regexString, htmlString, foundThis, key, category) {
+  const regX = new RegExp(regexString);
+  if (regX.test(htmlString)) {
+    if (isKeyUnique(foundThis, key)) {
+      switch (category) {
+        case '10':
+          if (isSupported(key)) {
+            foundThis.supported.analytics.push(key);
+          } else {
+            foundThis.notSupported.analytics.push(key);
+          }
+          break;
+        case '36':
+          if (isSupported(key)) {
+            foundThis.supported.ads.push(key);
+          } else {
+            foundThis.notSupported.ads.push(key);
+          }
+          break;
+      }
+    }
+  }
+}
+/**
+ * Checks to see if key is unique within the object
+ * @param {Object} obj - Object separating the 3p services by support
+ * @param {String} key - name of third party service
+ * @return {boolean} 
+ */
+function isKeyUnique(obj, key) {
+  const doesKeyExist = obj.supported.ads.includes(key) + 
+        obj.supported.analytics.includes(key) + 
+        obj.notSupported.ads.includes(key) + 
+        obj.notSupported.analytics.includes(key);
+  return !doesKeyExist;
+}
+/**
+ * TODO (alwalton@): get list of supported ads/analytics programatically
```

Comment:
When/how are we planning to do this? Is there a section about this in your design doc?